src/D2L.CodeStyle.Analyzers/Language/DefaultStructCreation/DefaultStructCreationAnalyzer.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace D2L.CodeStyle.Analyzers.Language.DefaultStructCreation {
+
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed partial class DefaultStructCreationAnalyzer : DiagnosticAnalyzer {
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+			=> ImmutableArray.Create( Diagnostics.DontCallDefaultStructConstructor );
+
+		public override void Initialize( AnalysisContext context ) {
+			context.EnableConcurrentExecution();
+
+			context.ConfigureGeneratedCodeAnalysis(
+				GeneratedCodeAnalysisFlags.None
+			);
+
+			context.RegisterOperationAction(
+				ctx => AnalyzeObjectCreation(
+					context: ctx,
+					operation: ctx.Operation as IObjectCreationOperation
+				),
+				OperationKind.ObjectCreation
+			);
+		}
+
+		private static void AnalyzeObjectCreation(

```

Comment:
If we want to be less mean there could be a whitelist of types this applies to (like the code fix)